Apps/W1/HybridGP/app/src/Migration/Support/WizardIntegration.codeunit.al

```diff
@@ -124,16 +123,26 @@ codeunit 4035 "Wizard Integration"
         exit(1);
     end;
 
+    procedure StartGPHistoricalJobMigrationAction(JobNotRanNotification: Notification)
+    begin
+        ScheduleGPHistoricalSnapshotMigration();
+    end;
+
     procedure ScheduleGPHistoricalSnapshotMigration()
     var
+        GPConfiguration: Record "GP Configuration";
         HybridCloudManagement: Codeunit "Hybrid Cloud Management";
+        UserPermissions: Codeunit "User Permissions";
         TimeoutDuration: Duration;
         MaxAttempts: Integer;
         QueueCategory: Code[10];
         IsHandled: Boolean;
         OverrideTimeoutDuration: Duration;
         OverrideMaxAttempts: Integer;
     begin
+        if not UserPermissions.IsSuper(UserSecurityId()) then
+            exit;
+
         TimeoutDuration := GetDefaultGPHistoricalMigrationJobTimeoutDuration();
```

Comment:
We are lacking the check if the user can schedule the tasks.
This can be done by checking:
  if not TaskScheduler.CanCreateTask() then 
      exit 
   if not JobQueueEntry.WritePermission then
     exit
     
     JobQueueEntry is a record